src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,209 @@
+                            str += Indentation + (i + 1) + "." + tasks.get(i).toString();
+                        }
+                    }
+                } catch (Exception e) {
+                    System.out.println(Indentation + Horizontal);
+                    System.out.println("  ☹ OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println(Indentation + Horizontal);
+
+                    str += "  ☹ OOPS!!! The description of a todo cannot be empty. \n";
+                }
+
+            } else if (words[0].equals("todo")) {
+                try {
+                    if (words[1].equals(null)) {
+                        System.out.println(Indentation + Horizontal);
+                        System.out.println("  ☹ OOPS!!! The description of a todo cannot be empty.");
+                        System.out.println(Indentation + Horizontal);
+                        str += "  ☹ OOPS!!! The description of a todo cannot be empty.";
+                    }
+                    info = command.substring(command.indexOf(" ") + 1);
+                    task = new Todo(info, false);
+                    tasks.add(task);
+                    str += new Todo(info, false);
+                } catch (Exception e) {
+                    System.out.println(Indentation + Horizontal);
+                    System.out.println("  ☹ OOPS!!! The description of a todo cannot be empty.");
+                    System.out.println(Indentation + Horizontal);
+                    str += "  ☹ OOPS!!! The description of a todo cannot be empty.";
+                }
+
+            } else if (words[0].equals("deadline")) {
+                try {
+                    info = command.substring(command.indexOf(" ") + 1, command.indexOf(" /by"));
+                    String deadline = command.substring(command.indexOf("/by") + 4);
+
+                    try {
+                        //Date and Time
+                        DateTimeFormatter formatter = DateTimeFormatter.ofPattern(("MM/dd/yyyy HHmm"));
+                        LocalDateTime datetime1 = LocalDateTime.parse(deadline, formatter);
+                        //System.out.println(datetime1.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")));
+
+                        task = new Deadline(info,
+                                datetime1.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")), false);
+                        str += new Deadline(info,
+                                datetime1.format(DateTimeFormatter.ofPattern("MMM d yyyy HHmm")), false);
+                        str += "/n";
+                    } catch (DateTimeParseException e) {
+                        //System.out.println(deadline);
+                        task = new Deadline(info, deadline, false);
+
+                        str += new Deadline(info, deadline, false);
+                        str += "/n";
+                    }
+                    tasks.add(task);
+                } catch (Exception e) {
```

Comment:
It appears that you have many try-catch blocks here. You may consider just using a single try-catch block and then throw the necessary exceptions (eg. `InvalidInputException`, `EmptyInputException`)  that will subsequently be handled in `Duke` as mentioned in my previous comment.